src/lib/operators/join_sort_merge.cpp

```diff
@@ -845,14 +1160,72 @@ class JoinSortMerge::JoinSortMergeImpl : public AbstractReadOnlyOperatorImpl {
     }
   }
 
+  bool _check_if_table_contains_null(const Table& table) {
+    const auto& table_column_definition =
+        table.column_definitions()[_sort_merge_join._primary_predicate.column_ids.second];
+    if (!table_column_definition.nullable) return false;
+
+    const auto chunk_count = table.chunk_count();
+    auto has_null = false;
+
+    for (auto cluster_id = ChunkID{0}; cluster_id < chunk_count; ++cluster_id) {
+      const auto& chunk = table.get_chunk(cluster_id);
+      const auto& segment = chunk->get_segment(_sort_merge_join._primary_predicate.column_ids.second);
+      segment_iterate<T>(*segment, [&](const auto& position) {
+        if (position.is_null()) {
+          has_null = true;
+          return;
+        }
+      });
+      if (has_null) return true;
+    }
+    return false;
+  }
+
  public:
   std::shared_ptr<const Table> _on_execute() override {
-    const auto include_null_left = (_mode == JoinMode::Left || _mode == JoinMode::FullOuter);
-    const auto include_null_right = (_mode == JoinMode::Right || _mode == JoinMode::FullOuter);
-    auto radix_clusterer = RadixClusterSort<T>(
-        _sort_merge_join.left_input_table(), _sort_merge_join.right_input_table(),
-        _sort_merge_join._primary_predicate.column_ids, _primary_predicate_condition == PredicateCondition::Equals,
-        include_null_left, include_null_right, _cluster_count, _performance);
+    if (_mode == JoinMode::Semi) {
```

Comment:
Can you add a comment that you're now going to check for early exit cases?

Further, I think we should remove the code for non-equal semi joins. It bloats the code is not going to occur in Hyrise any time soon.

Besides that, I don't like this current code too much. Too many lines of code for too simple checks. Can we somehow reduce that?

Reply:
I removed the `non-equal ` semi-join. Since GitHub is confusing me quite a bit right now, can you tell me again which part of the code you do not like at the moment?

Reply:
Hmm, never mind. A comment stating that we are going to check for simple early exists should be sufficient.

Reply:
I added a comment:

```
    // In some certain semi and anti join cases, we do not need to execute the sort-merge join algorithm. These can be
    // determined before the merge.
    // The case for the semi-join is:
    // - The right table is empty. We can return an empty left table.
    // The case for the anti null as false join is:
    // - The right table is empty. We can return the left input table.
    // The cases for the anti null as false join are:
    // - The right table is empty. We can return the left input table.
    // - If there is only one join predicate and the right table contains a null value in the join column, we can
    //   return an empty left table.
```